cmake/modules/snippets.cmake

```diff
@@ -94,8 +94,8 @@ function(zephyr_process_snippets)
   include(${snippets_generated})
 
   # Propagate include()d build system settings to the caller.
-  set(DTC_OVERLAY_FILE ${DTC_OVERLAY_FILE} PARENT_SCOPE)
-  set(OVERLAY_CONFIG ${OVERLAY_CONFIG} PARENT_SCOPE)
+  set(DTC_OVERLAY_FILE ${DTC_OVERLAY_FILE} CACHE STRING "DTC overlay" FORCE)
+  set(OVERLAY_CONFIG ${OVERLAY_CONFIG} CACHE STRING "Kconfig overlay" FORCE)
```

Comment:
afaict this will result in an ever expanding list whenever CMake re-runs, due to the fact that `snippets.py` generates `list(APPEND ...)` function calls.

Instead of trying to adjust this, which could collide with user provided values as well as value propagated from sysbuild, then I consider if a better option would be to keep the `DTC_OVERLAY_FILE` and `OVERLAY_CONFIG` for the user and have snippets specific variants.

Or maybe even better, decide where in `zephyr_get()` snippet controlled settings should be considered, just like we have for sysbuild controlled settings.
That would allow for clear order of precedence, both for single vars, as well as `MERGE`, ref: #56801 .

Then we could have follow order of precedence:
- sysbuild defined (which originates from user input, using `-D`)
- cache values (provided by user thrugh `-D`, thus taking precendence over snippets)
- snippets defined values
- env defined values
- local defined values

Reply:
I will try to see if I can rework this completely to define and use something like `snippet_conf_files` and `snippet_dts_files` (similar to what shield implementation does; i.e. `shield_conf_files` and `shield_dts_files`).